### PR TITLE
partition_manager: Add Zigbee support

### PR DESCRIPTION
**Pull request by mariuszpos**
Add Zigbee support in Partition Manager.

### CODE REVIEW

File: subsys/partition_manager/pm.yml.zboss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <autoconf.h>
+
+zboss_nvram:
+  placement: {after: [mcuboot_storage, app]}
```
Comment (hakonfam) [inline subsys/partition_manager/pm.yml.zboss:4]: `mcuboot_storage` is not in use any more, so this can be removed.

Comment (hakonfam) [inline subsys/partition_manager/pm.yml.zboss:4]: I assume that this should be page-aligned? If so consider adding an `align` property as well.

```suggestion
  placement: {after: [app], align: {start: DT_FLASH_ERASE_BLOCK_SIZE}}
```

Comment (mariuszpos) [reply subsys/partition_manager/pm.yml.zboss:4]: Good point.

**Comment by hakonfam** [comment]
You should squash some of your commits to ensure that the sample builds in all of the commits, this to retain bisectability. I would propose a single commit adding PM with cmake and .yml, then you can remove DT defines in the next commit. The test should also pass in all commits for the same reason.

**Comment by wbober** [comment]
@rlubos Any chance to get this merged? The CI fail doesn't seem to be releated.

**Comment by rlubos** [comment]
@wbober related or not, my merge button is grayed out.  I suggest a rebase.

**Comment by mariuszpos** [comment]
Rebased

**Comment by wbober** [comment]
@rlubos Success!

**Comment by rlubos** [comment]
Finally

**Comment by wbober** [comment]
Thanks a mil!

**Pull request merged**